File: problems/06_linked_list/0025_reverse_nodes_in_k_group/solution.dart
// Problem: 0025. Reverse Nodes in k-Group
// Difficulty: Hard

class ListNode {
  int val; ListNode? next;
  ListNode([this.val = 0, this.next]);
}

ListNode? fromList(List<int> a) {
  // TODO: implement
  return null;
}

List<int> toList(ListNode? h) {
  final out = <int>[];
  while (h != null) { out.add(h.val); h = h.next; }
  return out;
}

ListNode? reverseKGroup(ListNode? head, int k) {
  // TODO: implement
  return null;
}


void main() {
  // Test 1
  // final result = solve(...);
  print('All tests passed!');
}

File: problems/06_linked_list/0025_reverse_nodes_in_k_group/solution.py
"""
Problem: 0025. Reverse Nodes in k-Group
Difficulty: Hard
URL: https://leetcode.com/problems/reverse-nodes-in-k-group/

解法：dummy + 分組反轉，每組獨立反轉並串接，O(n)/O(1)。
"""

from __future__ import annotations

from typing import Optional

class ListNode:
    def __init__(self, val: int = 0, next: Optional["ListNode"] = None):
        self.val = val
        self.next = next

def to_list(h):
    out = []
    while h: out.append(h.val); h = h.next
    return out

def from_list(arr):
    d = ListNode(); c = d
    for v in arr: c.next = ListNode(v); c = c.next
    return d.next

class Solution:
    def reverseKGroup(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        pass


if __name__ == "__main__":
    solution = Solution()

    # Test 1
    # result = solution.solve(...)
    # assert result == expected

    print("All tests passed!")
